core/ir/synthetic/encoding_common.h

```diff
@@ -33,8 +33,84 @@
 #ifndef _SYNTHETIC_ENCODING_COMMON_H_
 #define _SYNTHETIC_ENCODING_COMMON_H_
 
+/* This synthetic ISA is a made up ISA that has the purpose of preserving register
+ * dependencies and giving hints on the type of operations each instruction is performing.
+ * For this reason the majority of operations that would normally work on instructions
+ * coming from an actual ISA are not supported.
+ * The only operations that we support are instr_encode() and decode() when DynamoRIO's
+ * host is an x86 architecture.
+ * Also note that in DynamoRIO we use the #instr_t ISA mode to determine what type of
+ * encoding to perform, and #dr_context_t ISA mode to determine what type of decoding to
+ * perform.
+ * Currently the only exception to this rule happens for decoding of synthetic
+ * instructions, where #instr_t ISA mode takes precedence and #dr_context_t ISA mode is
+ * ignored if #instr_t ISA mode is DR_ISA_SYNTHETIC.
+ * XXX i#1684: Note that this is part of a larger issue, where lack of cross-arch support
+ * in the same build of DynamoRIO is limiting us.
+ */
+
+/* Here we describe the encoding scheme for the Synthetic ISA that is enforced in decode.c
+ * and encode.c.
+ *
+ * Encoded instructions are 4 bytes aligned.
+ *
+ * All instruction encodings begin with the following 4 bytes, which follow this scheme:
+ * |----------------------| |--| |----| |----|
+ * 31..               ..10  9,8   7..4   3..0
+ *       category          eflags #src   #dst
+ *
+ * 22 bits, category: it's a high level representation of the opcode of an instruction.
+ * Each bit represents one category following #dr_instr_category_t.
+ * Note that an instruction can belong to more than one category, hence multiple bits can
+ * be set.
+ * 2 bits, eflags: most significant bit set to 1 indicates the instruction reads at least
+ * one arithmetic flag, least significant bit set to 1 indicates the instruction writes
+ * at least one arithmetic flag.
+ * 4 bits, #src: number of source operands (read) that are registers.
+ * 4 bits, #dst: number of destination operands (written) that are registers.
+ * (Note that we are only interested in register dependencies, hence operands that are
+ * not registers, such as immediates or memory references, are not present.)
```

Comment:
Clarify that registers used to compute addresses are present as source operands.

Reply:
Done.